File: java/socketd/src/main/java/org/noear/socketd/transport/client/ClientHandshakeResult.java
package org.noear.socketd.transport.client;

import org.noear.socketd.transport.core.ChannelInternal;

/**
 * 客户端握手结果
 *
 * @author noear
 * @since 2.0
 */
public class ClientHandshakeResult {
    private final ChannelInternal channel;
    private final Throwable throwable;

    public ChannelInternal getChannel() {
        return channel;
    }

    public Throwable getThrowable() {
        return throwable;
    }

    public ClientHandshakeResult(ChannelInternal channel, Throwable throwable) {
        this.channel = channel;
        this.throwable = throwable;
    }
}

File: javascript/socketd-examples/demo01-js-ws-esm/js/socketd/transport/client/ClientHandshakeResult.js
/**
 * 客户端握手结果
 *
 * @author noear
 * @since 2.0
 */
export class ClientHandshakeResult {
    constructor(channel, throwable) {
        this._channel = channel;
        this._throwable = throwable;
    }
    getChannel() {
        return this._channel;
    }
    getThrowable() {
        return this._throwable;
    }
}
